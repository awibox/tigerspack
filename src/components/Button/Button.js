import React from 'react';
import PropTypes from 'prop-types';

const ButtonStyle = {
  borderRadius: '4px',
  boxSizing: 'border-box',
  position: 'relative',
  fontSize: '14px',
  outline: 'none',
  transition: 'all 0.1s',
  display: 'inline-block',
  marginBottom: 0,
  fontWeight: 500,
  textAlign: 'center',
  verticalAlign: 'middle',
  cursor: 'pointer',
  backgroundImage: 'none',
  border: '1px solid transparent',
  whiteSpace: 'nowrap',
  padding: '6px 16px',
  lineHeight: 1.52857143,
  userSelect: 'none',
  backgroundColor: 'rgba(61, 94, 97, .9)',
  color: '#fff',
  ':hover': {
    textDecoration: 'none',
    backgroundColor: 'rgba(61, 94, 97, .8)',
  },
};

const Button = (props) => {
  const { children, onClick } = props;
  return (
    <div style={ButtonStyle} onClick={onClick}>
      {children}
    </div>
  );
};

Button.propTypes = {
  children: PropTypes.any.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default Button;
