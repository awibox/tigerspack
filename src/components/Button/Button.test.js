import React from 'react';
import { shallow } from 'enzyme';
import Button from '.';

const clickFn = jest.fn();

describe('Button', () => {
  it('should render correctly', () => {
    const component = shallow(<Button onClick={clickFn}>Test button</Button>);
    expect(component).toMatchSnapshot();
  });
  it('should call clickFn on button click', () => {
    const component = shallow(<Button onClick={clickFn}>Test button</Button>);
    component.simulate('click');
    expect(clickFn).toHaveBeenCalled();
  });
});
