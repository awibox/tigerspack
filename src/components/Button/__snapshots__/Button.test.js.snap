// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<div
  onClick={[MockFunction]}
  style={
    Object {
      ":hover": Object {
        "backgroundColor": "rgba(61, 94, 97, .8)",
        "textDecoration": "none",
      },
      "backgroundColor": "rgba(61, 94, 97, .9)",
      "backgroundImage": "none",
      "border": "1px solid transparent",
      "borderRadius": "4px",
      "boxSizing": "border-box",
      "color": "#fff",
      "cursor": "pointer",
      "display": "inline-block",
      "fontSize": "14px",
      "fontWeight": 500,
      "lineHeight": 1.52857143,
      "marginBottom": 0,
      "outline": "none",
      "padding": "6px 16px",
      "position": "relative",
      "textAlign": "center",
      "transition": "all 0.1s",
      "userSelect": "none",
      "verticalAlign": "middle",
      "whiteSpace": "nowrap",
    }
  }
>
  Test button
</div>
`;
